docs/source/about_mapstyle_vs_iterable.mdx

```diff
@@ -0,0 +1,220 @@
+Thanks to memory mapping and the fact that Arrow is an in-memory format, reading data from disk doesn't do expensive system calls and deserialization.
+
+It also provides even faster data loading when iterating using a `for` loop by iterating on contiguous Arrow record batches.
+
+However as soon as your [`Dataset`] has an indices mapping (via [`Dataset.shuffle`] for example), the speed can become 10x slower.
+This is because there is an extra step to get the row index to read using the indices mapping, and most importantly because you no longer read contiguous chunks of data.
+To get the optimal speed back, you'd need to rewrite the entire dataset on your disk again using `Dataset.flatten_indices` which removes the indices mapping.
+This may take a lot of time depending of the size of your dataset though:
+
+```python
+my_dataset[0]  # fast
+my_dataset = my_dataset.shuffle(seed=42)
+my_dataset[0]  # up to 10x slower
+my_dataset = my_dataset.flatten_indices()  # rewrite the shuffled dataset on disk as contiguous chunks of data
+my_dataset[0]  # fast again
+```
+
+
+In this case is it recommended to switch to an [`IterableDataset`] and leverage its fast approximate shuffling method [`IterableDataset.shuffle`].
+It only shuffles the shards order and adds a shuffle buffer to your dataset, which keeps the speed of your dataset optimal.
+You can also reshuffle the dataset easily:
+
+```python
+for example in enumerate(my_iterable_dataset):  # fast
+    pass
+
+shuffled_iterable_dataset = my_iterable_dataset.shuffle(seed=42, buffer_size=100)
+
+for example in enumerate(shuffled_iterable_dataset):  # as fast as before
+    pass
+
+shuffled_iterable_dataset = my_iterable_dataset.shuffle(seed=1337, buffer_size=100)  # reshuffling using another seed is instantaneous
+
+for example in enumerate(shuffled_iterable_dataset):  # still as fast as before
+    pass
+```
+
+Note if you use your dataset on multiple epochs:
+
+The effective seed to shuffle the shards order and for the shuffle buffer is `seed + epoch`.
+It makes it easy to reshuffle a dataset betwen epochs:
+```python
+for epoch in range(n_epochs):
+    my_iterable_dataset.set_epoch(epoch)
+    for example in my_iterable_dataset:  # fast + reshuffled at each epoch using `effective_seed = seed + epoch`
+        pass
+```
+
+## Switch from map-style to iterable
+
+If you want to benefit from the "lazy" behavior of iterable datasets or their speed advantages, you can switch your map-style [`Dataset`] to an [`IterableDataset`]:
+```python
+my_iterable_dataset = my_dataset.to_iterable()
+```
+
+If you want to shuffle your dataset or [use it with a PyTorch DataLoader](./use_with_pytorch#stream-data), it is recommended to generate a sharded iterable dataset:
+```python
+my_iterable_dataset = my_dataset.to_iterable(num_shards=1024)
+my_iterable_dataset.n_shards  # 1024
+```
+
+Feel free to open a discussion on the 🤗 Datasets [forum](https://discuss.huggingface.co/c/datasets/10) if you have questions !
```

Comment:
```suggestion
Feel free to open a discussion on the 🤗 Datasets [forum](https://discuss.huggingface.co/c/datasets/10) if you have questions!
```

Reply:
I would remove this sentence altogether. Two existing links in our docs are more than enough  :).